overlord/hookstate/ctlcmd/umount.go

```diff
@@ -0,0 +1,81 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package ctlcmd
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/i18n"
+	"github.com/snapcore/snapd/systemd"
+)
+
+var (
+	shortUmountHelp = i18n.G("Undo a temporary or permanent mount")
+	longUmountHelp  = i18n.G(`
+The umount command unmounts the given mount point, which must have been
+previously created with "snapctl mount".`)
+)
+
+func init() {
+	addCommand("umount", shortUmountHelp, longUmountHelp, func() command { return &umountCommand{} })
+}
+
+type umountCommand struct {
+	baseCommand
+	Positional struct {
+		Where string `positional-arg-name:"<where>" required:"yes" description:"path to the mount point"`
+	} `positional-args:"yes" required:"yes"`
+}
+
+func (m *umountCommand) Execute([]string) error {
+	context, err := m.ensureContext()
+	if err != nil {
+		return err
+	}
+
+	if m.Positional.Where == "" {
```

Comment:
It doesn't seem so; at least, the unit test at https://github.com/mardy/snapd/blob/mount-control-3/overlord/hookstate/ctlcmd/umount_test.go#L108-L111 is able to trigger this condition.

Reply:
The test isn't triggering the go-flags validation because of the empty string arg (I left a comment in the test). Fixing that means we can also remove this